### PR TITLE
Prepare core libraries for July 2024 Releases

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/core/azure-core-experimental/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-### Features Added
-
-### Breaking Changes
+### Other Changes
 
-### Bugs Fixed
+#### Dependency Updates
 
-### Other Changes
+- Upgraded `azure-core` from `1.49.0` to `1.49.2`.
```
Comment (mssfang) [inline sdk/core/azure-core-experimental/CHANGELOG.md:9]:
```suggestion
- Upgraded `azure-core` from `1.49.0` to `1.50.0`.
```

File: sdk/core/azure-core-http-jdk-httpclient/CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-### Features Added
-
-### Breaking Changes
+### Other Changes
 
-### Bugs Fixed
+#### Dependency Updates
 
-### Other Changes
+- Upgraded `azure-core` from `1.49.0` to `1.49.2`.
```
Comment (mssfang) [inline sdk/core/azure-core-http-jdk-httpclient/CHANGELOG.md:9]: same here for the azure core version.

File: sdk/core/azure-core-http-vertx/CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-### Features Added
-
-### Breaking Changes
+### Other Changes
 
-### Bugs Fixed
+#### Dependency Updates
 
-### Other Changes
+- Upgraded `azure-core` from `1.49.1` to `1.49.2`.
```
Comment (mssfang) [inline sdk/core/azure-core-http-vertx/CHANGELOG.md:9]: same here for the azure core version

File: sdk/core/azure-core-management/CHANGELOG.md [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
-### Features Added
-
-### Breaking Changes
+### Other Changes
 
-### Bugs Fixed
+#### Dependency Updates
 
-### Other Changes
+- Upgraded `azure-core` from `1.49.1` to `1.49.2`.
```
Comment (mssfang) [inline sdk/core/azure-core-management/CHANGELOG.md:9]: same here, there are more places in the otherCHANGELOG.md.

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 5]
```diff
@@ -499,6 +499,18 @@
           "old" : ".*? com\\.azure\\.communication\\.chat\\.models.*",
           "new" : ".*? com\\.azure\\.communication\\.chat\\.models.*",
           "justification": "Migration to azure-json"
+        },
+        {
+          "ignore": true,
+          "code": "java.method.added",
+          "new": "method void com.azure.core.credential.AccessToken::<init>(java.lang.String, java.time.OffsetDateTime, java.time.OffsetDateTime)",
+          "justification": "ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE"
```
Comment (mssfang) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:507]: don't forget to update the justification message for the reason

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 6]
```diff
@@ -499,6 +499,18 @@
           "old" : ".*? com\\.azure\\.communication\\.chat\\.models.*",
           "new" : ".*? com\\.azure\\.communication\\.chat\\.models.*",
           "justification": "Migration to azure-json"
+        },
+        {
+          "ignore": true,
+          "code": "java.method.added",
+          "new": "method void com.azure.core.credential.AccessToken::<init>(java.lang.String, java.time.OffsetDateTime, java.time.OffsetDateTime)",
+          "justification": "Added getRefreshAt method to AccessToken class."
```
Comment (mssfang) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:507]: should be the constructor,

File: sdk/core/azure-core-amqp/CHANGELOG.md [hunk 7]
```diff
@@ -1,14 +1,9 @@
 # Release History
 
-## 2.10.0-beta.1 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
+## 2.9.7 (2024-07-11)
```
Comment (billwert) [inline sdk/core/azure-core-amqp/CHANGELOG.md:3]: maybe re-run the prep script to fix all the dates to today?

**Review by billwert — APPROVED** [review]
lgtm after the dates in the changelog are fixed. :)

**Review by anuchandy — APPROVED** [review]

**Review by mssfang — APPROVED** [review]

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 8]
```diff
@@ -499,6 +499,18 @@
...
+          "ignore": true,
+          "code": "java.method.added",
+          "new": "method void com.azure.core.credential.AccessToken::<init>(java.lang.String, java.time.OffsetDateTime, java.time.OffsetDateTime)",
+          "justification": "Added a new constructor to the AccessToken class."
+        },
+        {
+          "ignore": true,
+          "code": "java.method.added",
+          "new": "method java.time.OffsetDateTime com.azure.core.credential.AccessToken::getRefreshAt()",
+          "justification": "Added getRefreshAt method to AccessToken class."
```
Comment (srnagar) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:513]: Adding new methods in a minor version update is not a breaking change. Why is revapi suppression needed here?

**Review by srnagar — APPROVED** [review]

**Pull request merged**